test/com/linkedin/drelephant/spark/fetchers/SparkRestClientTest.scala

```diff
@@ -70,12 +74,18 @@ class SparkRestClientTest extends AsyncFunSpec with Matchers {
       sparkRestClient.fetchData(FetchClusterModeDataFixtures.APP_ID) map { restDerivedData =>
         restDerivedData.applicationInfo.id should be(FetchClusterModeDataFixtures.APP_ID)
         restDerivedData.applicationInfo.name should be(FetchClusterModeDataFixtures.APP_NAME)
-        restDerivedData.jobDatas should not be(None)
-        restDerivedData.stageDatas should not be(None)
-        restDerivedData.executorSummaries should not be(None)
+        restDerivedData.jobDatas should not be (None)
+        restDerivedData.stageDatas should not be (None)
+        restDerivedData.executorSummaries should not be (None)
+        restDerivedData.logDerivedData should be(None)
+      } flatMap {
+        case assertion: Try[Assertion] => assertion
+        case _ =>
+          sparkRestClient.fetchData(FetchClusterModeDataFixtures.APP_ID, fetchLogs = true)
+            .map { _.logDerivedData should not be(None) }
```

Comment:
Wouldn't that be yet another test for `SparkLogClient`? because `fetchData` doesn't do any log parsing, it simply uncompresses the logs, which is already covered by the current test.